### PR TITLE
feat: add CI

### PR DESCRIPTION
**Pull request by vargaadam**
https://github.com/emartech/google-application-platform/issues/765

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,8 @@
...
+COPY ./package*.json ./
 RUN npm ci
+COPY . .
 
-COPY . . 
-
-CMD ["npm", "start"]
\ No newline at end of file
+FROM build as test
+CMD ["npm", "run", "test"]
```
Comment (geritol) [inline Dockerfile:8]: Currently the published images default command is test, which is not good i think.
I would add an other stage thats cmd is npm start and publish that to the registry.

File: cloudbuild-production.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    args:
+      ["build", "--target", "test", "-t", "secret-backup-operator-test", "."]
+  - name: "secret-backup-operator-test"
+    dir: "/usr/secret-backup-operator"
+  - name: "gcr.io/cloud-builders/docker"
+    args:
+      [
+        "build",
+        "-t",
+        "gcr.io/ems-gap-images/secret-backup-operator:$TAG_NAME",
```
Comment (kissmikijr) [inline cloudbuild-production.yaml:12]: right now we use this image from eu.gcr.io 
https://github.com/emartech/google-application-platform/blob/1e530051753738ccf8363a4b79f06a9e974bb8ca/core/components/admin/secret-backup-operator/deployment.yaml#L50

File: cloudbuild-production.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        ".",
+      ]
+images: ["gcr.io/ems-gap-images/secret-backup-operator:$TAG_NAME"]
```
Comment (kissmikijr) [inline cloudbuild-production.yaml:15]: same as above

File: Dockerfile [hunk 4]
```diff
@@ -1,8 +1,11 @@
...
+COPY . .
 
-COPY . . 
+FROM build AS test
+CMD ["npm", "run", "test"]
 
-CMD ["npm", "start"]
\ No newline at end of file
+FROM build AS prod
+CMD ["npm", "run"]
```
Comment (kissmikijr) [inline Dockerfile:11]:
```suggestion
CMD ["npm", "run", "start"]
```

**Review by kissmikijr — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 4 above)
Comment (vargaadam) [reply Dockerfile:11]: 🙈

Comment (geritol) [reply Dockerfile]: or `npm start`

File: Dockerfile [hunk 5]
```diff
@@ -1,8 +1,11 @@
...
+COPY . .
 
-COPY . . 
+FROM build AS test
+CMD ["npm", "run", "test"]
 
-CMD ["npm", "start"]
\ No newline at end of file
+FROM build AS prod
+CMD ["npm", "run", "start"]
```
Comment (geritol) [inline Dockerfile:11]:
```suggestion
CMD ["npm", "start"]
```

**Review by geritol — APPROVED** [review]

**Review by kissmikijr — APPROVED** [review]

**Pull request merged**